### PR TITLE
integration test v2: mvcc resolve lock gc test

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Ref #12842

What's Changed:

```commit-message
mvcc resolve lock gc test for v2
```
This PR resuse test_mvcc_resolve_lock_gc_and_delete for raftstore-v2.
The PR also fixes a bug when reporting persisted/committed log duration which may cause panic:
- It may miss updating metrics of the porposal with index `to`, making relevant metrics of the proposal unintialized. Later in `write_write_detail`, it will panic due to `0 - a posive number`.

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore-v2/src/operation/ready/mod.rs [hunk 1]
```diff
@@ -680,7 +680,7 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             return;
         }
         let now = Instant::now();
-        for i in from + 1..to {
+        for i in from + 1..=to {
```
Comment (bufferflies) [inline components/raftstore-v2/src/operation/ready/mod.rs:683]: I think it belongs the type/bug, you should add some comment on the pr.

Comment (SpadeA-Tang) [reply components/raftstore-v2/src/operation/ready/mod.rs:683]: Done

File: components/test_raftstore-v2/src/server.rs [hunk 2]
```diff
@@ -634,6 +636,11 @@ impl ServerCluster {
         Ok(node_id)
     }
 
+    /// To trigger GC manually.
```
Comment (bufferflies) [inline components/test_raftstore-v2/src/server.rs:639]: remove the useless comment.

**Review by bufferflies — APPROVED** [review]

**Review by tonyxuqqi — APPROVED** [review]

File: components/raftstore-v2/src/operation/ready/mod.rs [hunk 3]
```diff
@@ -680,7 +680,7 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             return;
```
Comment (tabokie) [inline components/raftstore-v2/src/operation/ready/mod.rs:680]: update the argument name (from, to) to (old_index, new_index).

**Review by tabokie — APPROVED** [review]

File: components/test_raftstore-v2/src/server.rs (hunk 2 above)
Comment (SpadeA-Tang) [reply components/test_raftstore-v2/src/server.rs]: Done

File: components/raftstore-v2/src/operation/ready/mod.rs (hunk 3 above)
Comment (SpadeA-Tang) [reply components/raftstore-v2/src/operation/ready/mod.rs:680]: Done

**Pull request merged**